src/dbnode/storage/database.go

```diff
@@ -571,8 +572,9 @@ func (d *db) Terminate() error {
 }
 
 func (d *db) Close() error {
-	// NB(bodu): Wait for fs processes to finish.
-	d.mediator.WaitForFileSystemProcesses()
+	// NB(bodu): Disable file ops waits for current fs processes to
+	// finish before disabling.
+	d.mediator.DisableFileOps()
```

Comment:
Will it still wait? Or just it just disable it?

Reply:
It still waits, now it waits for both fs manager and cold flush manager to finish.
```
status := m.databaseFileSystemManager.Disable()
for status == fileOpInProgress {
        m.sleepFn(fileOpCheckInterval)
        status = m.databaseFileSystemManager.Status()
}
// Even though the cold flush runs separately, its still
// considered a fs process.
status = m.databaseColdFlushManager.Disable()
for status == fileOpInProgress {
        m.sleepFn(fileOpCheckInterval)
        status = m.databaseColdFlushManager.Status()
}
```

Reply:
👍  one super nit could be to change the name to make it explicit to something like "DisableFileOpsAndWait()"?